r/configure.win

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+# generate code
+if [ "$ARROW_R_DEV" == "TRUE" ]; then
+  echo "*** Generating code with data-raw/codegen.R"
+  "${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" data-raw/codegen.R
+fi
+
+VERSION=$(grep ^Version DESCRIPTION | sed s/Version:\ //)
+"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "tools/winlibs.R" $VERSION
+
+if [ $? -ne 0 ]; then
```

Comment:
Is this expected? And should you exit immediately if this failure is encountered.

Reply:
`tools/winlibs.R` will exit nonzero if https://github.com/apache/arrow/pull/4538/files/328bc376f9adb2717d1505f96d8d7452ee9d8212#diff-48db18073973d5ea65a7f0023f920145R22 fails to find and download a prebuilt binary. This triggers the fallback build-package-without-libarrow behavior (see https://github.com/apache/arrow/pull/4471).

I can add some comments to explain.

Reply:
Ok now I understand more clearly. I think this warrants a comment or clearer message, e.g.

"Failed to download and extract libarrow runtime and dependencies. The R arrow package will compile with shim methods. Please run arrow::install_arrow() to install required runtime libraries and recompile this package with proper support."

Reply:
Or separate this message in half, one for users, and a comment for maintainers.

Reply:
Just pushed some comments